client/src/index.jsx

```diff
@@ -1,4 +1,157 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import styled from 'styled-components';
+import $ from 'jquery';
+import Reviews from './components/Reviews.jsx';
+import RatingDetails from './components/RatingDetails.jsx';
+import Filter from './components/Filter.jsx'
+import AddReview from './components/AddReview.jsx'
 
-ReactDOM.render(<App />, document.getElementById('app'))
\ No newline at end of file
+const Container = styled.div`
+  float: left;
+  width: 625px;
+  padding-right: 10px;
+  padding-left: 8px;
+`;
+
+const StyledLink = styled.a`
+  color: #00635d;
+  textDecoration: none;
+  font-family: Lato, Helvetica Neue, Helvetica, sans-serif;
+  cursor: pointer;
+  &:hover {text-decoration: underline};
+  display: inline-block;
+  position: relative;
+  flex-basis: 400px;
+`;
+
+const Search = styled.span`
+  float: right;
+  color: #333333;
+  border: #DCD6CC 1px solid;
+  border-radius: 3px;
+  background: #FFFFFF;
+  resize: none
+`;
+
+const Align = styled.div`
+  display: flex;
+  justify-content: space-between;
+  padding: 0;
+  margin: 10px;
+
+`;
+
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      reviews: [],
+      users: [],
+      ratings: [],
+      rating: 0,
+      ratedReviews: [],
+      // five: 0,
+      // four: 0,
+      // three: 0,
+      // two: 0,
+      // one: 0,
+    };
+    this.handleReviews = this.handleReviews.bind(this);
+  }
+
+  // componentDidMount() {
+  //   fetch(this.getAllReviews())
+  //     .then(() => {
+  //       this.sortByRating();
+  //     })
+  //     .then(() => {
+  //       this.getAllUsers();
+  //     });
+  // }
+
+  componentDidMount() {
+    this.getAllReviews();
+    this.getAllUsers();
+  }
+
+  // IMPLEMENT REACT ROUTING
+  getAllReviews() {
+    $.get('/books/1/reviews', (data) => {
+      const ratings = [];
+      data.map((review) => { ratings.push(review.rating); });
```

Comment:
you could possibly use the array method **.reduce**
to reduce the data to a tuple with the following information:
**[coutOfAllReviews, totalSumOfAllReviews]**

By using reduce you could iterate over the array just once to get an average and total count.